File: python/yax/test/dap4saxtest.py
# Copyright 2013, UCAR/Unidata.
#  See the COPYRIGHT file for more information.

import os
import sys
import getopt
import codecs
import xml.sax
import StringIO

sys.path.append(os.path.join(os.getcwd(),"yax"))

from saxevent import SaxEvent
from saxeventhandler import SaxEventHandler
import saxeventtype
import util
from Dap4SaxParser import Dap4SaxParser
from dap4saxeventhandler import Dap4SaxEventHandler

def main() :
  flags = util.FLAG_NONE;

  try:
    opts, args = getopt.getopt(sys.argv[1:], "wleTd", ["help", "output="])
  except getopt.GetoptError as err:
    # print help information and exit:
    print(err) # will print something like "option -a not recognized"
    sys.exit(1)

  if (len(args) == 0) :
    sys.stderr.write("no input\n")
    sys.exit(1)

  debuglevel = 0
  flags = util.FLAG_NOCR # always
  for o, a in opts:
    if o == "-w":
      flags = flags + util.FLAG_TRIMTEXT
    elif o == "-l":
      flags = flags + util.FLAG_ELIDETEXT
    elif o == "-e":
      flags = flags + util.FLAG_ESCAPE
    elif o == "-d":
      debuglevel = 1

  # push parser
  dap4pushparser = Dap4SaxParser()
  if (debuglevel > 0) :
    dap4pushparser.setDebugLevel(1);

  # read the document as utf
  document = codecs.open(args[0], encoding='utf-8').read()

  # Create handler
  handler = Dap4SaxEventHandler(document,dap4pushparser)

  # Parse the document
  handler.parse()

  sys.stdout.write("No error\n");
  sys.exit(0);
# end main()

if __name__ == '__main__':
  main()

File: java/src/yax/test/Dap4SaxTest.java
package yax.test;

import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.CommandLineParser;
import org.apache.commons.cli.Options;
import org.apache.commons.cli.PosixParser;
import org.w3c.dom.Node;
import yax.lex.*;

import java.io.FileReader;
import java.io.IOException;
import java.nio.charset.Charset;

public abstract class Dap4SaxTest
{

    //////////////////////////////////////////////////
    // Constants

    //////////////////////////////////////////////////

    static public void
    main(String[] argv)
    {
        int flags = Util.FLAG_NONE;
        SaxEventType tokentype = null;
        Node[] nodep = new Node[]{null};

        String document;
        int i, c;
        Dap4SaxParser dap4pushparser = null;
        Dap4SaxEventHandler dap4eventhandler = null;

        try {
            Options options = new Options();
            options.addOption("w", false, "trim whitespace ");
            options.addOption("l", false, "Limit size of text printout");
            options.addOption("e", false, "Escape control characters");
            options.addOption("t", false, "Trace parser");
            options.addOption("d", false, "Parser debug");
            options.addOption("T", false, "Trace lexer");

            CommandLineParser clparser = new PosixParser();
            CommandLine cmd = clparser.parse(options, argv);

            argv = cmd.getArgs();

            if(argv.length == 0) {
                System.err.printf("no input\n");
                System.exit(1);
            }

            document = getinput(argv[0]);

            flags = Util.FLAG_NOCR; // always
            if(cmd.hasOption('w'))
                flags |= Util.FLAG_TRIMTEXT;
            if(cmd.hasOption('l'))
                flags |= Util.FLAG_ELIDETEXT;
            if(cmd.hasOption('e'))
                flags |= Util.FLAG_ESCAPE;
            if(cmd.hasOption('T'))
                flags |= Util.FLAG_TRACE;

            // 1. push parser
            dap4pushparser = new Dap4SaxParser();
            if(cmd.hasOption('d'))
                dap4pushparser.setDebugLevel(1);

            // 2. event handler
            dap4eventhandler = new Dap4SaxEventHandler(document, dap4pushparser);
            if(!dap4eventhandler.parse()) {
                System.err.println("Parse failed");
            }
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }
        System.err.println("No error");
        System.exit(0);
    }

    static String
    getinput(String filename)
        throws IOException
    {
        StringBuilder buf = new StringBuilder();
        FileReader file = new FileReader(filename);
        int c;

        while((c = file.read()) > 0) {
            buf.append((char) c);
        }
        return buf.toString();
    }
} // class Dap4SaxTest




